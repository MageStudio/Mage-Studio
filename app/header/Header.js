import Link from 'next/link'
import React from 'react';
import { Menu, Dropdown, Icon, message, Col, Row, Divider } from 'antd';

import ProjectMenu from './menu/ProjectMenu';
import SceneMenu from './menu/SceneMenu';

import Logo from './Logo';
import './header.scss';

const onClick = ({ key }) => {
    message.info(`Click on item ${key}`);
};

const menu = (
    <Menu onClick={onClick}>
        <Menu.Item className="option" key="1">1st menu item</Menu.Item>
        <Menu.Item className="option" key="2">2nd memu item</Menu.Item>
        <Divider/>
        <Menu.Item className="option" key="3">3rd menu item</Menu.Item>
    </Menu>
);

const Header = () => (
    <Row className="header">
        <Col span={12}>
            <Logo />
            <ProjectMenu />
            <SceneMenu />
            <Dropdown overlay={menu}>
                <a className="ant-dropdown-link" className="main-menu-item" href="#">
                    View
                </a>
            </Dropdown>
            <Dropdown overlay={menu}>
                <a className="ant-dropdown-link" className="main-menu-item" href="#">
                    Help
                </a>
            </Dropdown>
        </Col>
    </Row>
);

export default Header;
