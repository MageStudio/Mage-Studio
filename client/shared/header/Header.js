import React from 'react';
import { connect } from 'react-redux';

import { requestSceneJson } from '../../editor/actions/scene';
import { saveScript } from '../../editor/actions/scripts';

import HelpMenu from './menu/HelpMenu';
import ViewMenu from './menu/ViewMenu';
import FileMenu from './menu/FileMenu';

import Logo from './Logo';
import style from './header.module.scss';

export const Header = ({
    isCodeEditor,
    onNewScript,
    onSaveScene,
    onSaveScript,
    scripts,
    config }) => {

    const handleSaveScript = () => {
        const { editor: { filename, code, scriptType } } = scripts;
        const { project } = config;

        onSaveScript(project, filename, code, scriptType);
    };

    return (
        <div className={style.header}>
            <Logo />
            <FileMenu
                isCodeEditor={isCodeEditor}
                onNewScript={onNewScript}
                onSaveScene={onSaveScene}
                onSaveScript={handleSaveScript} />
            <ViewMenu />
            <HelpMenu />
        </div>
    );
};

const mapStateToprops = ({ config, scripts }) => ({
    config,
    scripts
});

const mapDispatchToProps = (dispatch) => ({
    onSaveScene: () => dispatch(requestSceneJson()),
    onSaveScript: (project, filename, code, type) => dispatch(saveScript(project, filename, code, type))
});

export default connect(mapStateToprops, mapDispatchToProps)(Header);
