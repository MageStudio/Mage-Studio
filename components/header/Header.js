import Link from 'next/link'
import React from 'react';
import { Menu, Dropdown, Icon, message, Col, Row, Divider } from 'antd';
import './header.scss';

const style = {
    marginRight: 24
};

const onClick = ({ key }) => {
    message.info(`Click on item ${key}`);
};

const menu = (
    <Menu onClick={onClick} theme={"dark"}>
        <Menu.Item key="1">1st menu item</Menu.Item>
        <Menu.Item key="2">2nd memu item</Menu.Item>
        <Divider/>
        <Menu.Item key="3">3rd menu item</Menu.Item>
    </Menu>
);

const Header = () => (
    <Row className="header">
        <Col span={12}>
            <Dropdown overlay={menu}>
                <a className="ant-dropdown-link" className="main-menu-item" href="#">
                    Project
                </a>
            </Dropdown>
            <Dropdown overlay={menu}>
                <a className="ant-dropdown-link" className="main-menu-item" href="#">
                    Scene
                </a>
            </Dropdown>
            <Dropdown overlay={menu}>
                <a className="ant-dropdown-link" className="main-menu-item" href="#">
                    View
                </a>
            </Dropdown>
            <Dropdown overlay={menu}>
                <a className="ant-dropdown-link" className="main-menu-item" href="#">
                    Help
                </a>
            </Dropdown>
        </Col>
    </Row>
);

export default Header;
