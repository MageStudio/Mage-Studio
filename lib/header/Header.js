import React from 'react';
import { Menu, Dropdown, message, Divider } from 'antd';

import ProjectMenu from './menu/ProjectMenu';
import SceneMenu from './menu/SceneMenu';

import Logo from './Logo';
import './header.scss';

const onClick = ({ key }) => {
    message.info(`Click on item ${key}`);
};

const menu = (
    <Menu onClick={onClick}>
        <Menu.Item className="option" key="1">1st menu item</Menu.Item>
        <Menu.Item className="option" key="2">2nd memu item</Menu.Item>
        <Divider/>
        <Menu.Item className="option" key="3">3rd menu item</Menu.Item>
    </Menu>
);

const Header = ({ isCodeEditor }) => (
    <div className="header">
        <Logo />
        <ProjectMenu
            isCodeEditor={isCodeEditor} />
        <SceneMenu />
        <Dropdown overlay={menu}>
            <span className="main-menu-item" href="#">
                View
            </span>
        </Dropdown>
        <Dropdown overlay={menu}>
            <span className="main-menu-item" href="#">
                Help
            </span>
        </Dropdown>
    </div>
);

export default Header;
