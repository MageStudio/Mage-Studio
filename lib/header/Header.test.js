import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Header from './Header';

describe('Header', () => {

    it('should render fine', () => {
        const component = shallow(<Header/>);
        const json = toJSON(component);

        expect(json).toMatchSnapshot();
    });
});