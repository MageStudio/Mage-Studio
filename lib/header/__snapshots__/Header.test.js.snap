// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render fine 1`] = `
<div
  className="header"
>
  <Logo />
  <FileMenu
    onSaveScript={[Function]}
  />
  <ViewMenu />
  <HelpMenu />
</div>
`;
